main.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+import os.path
+from argparse import Namespace
+
+ERR_MSG = "Error message: Invalid input. Make sure the input file is a valid csv file."
+
+
+def get_args() -> Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--input_file", type=str, required=True)
+    parser.add_argument("-o", "--output_file", type=str, required=True)
+    parser.add_argument("-r", "--row_limit", type=int, required=True)
+    return parser.parse_args()
+
+
+def is_input_valid(input_file: str, row_limit: int) -> bool:
+    is_valid_file = os.path.isfile(input_file) and os.path.exists(input_file)
+    is_csv = input_file.lower().endswith(".csv")
+    if not (is_valid_file and is_csv):
+        return False
+
+    with open(input_file, "r") as fp:
+        num_of_lines = len(fp.readlines())
+        return num_of_lines > row_limit
+
+
+def get_file_chunks(input_file: str, row_limit: int) -> list[list[str]]:
```

Comment:
This object, `list[list[str]]`, is a bit cumbersome and doesn't really mean much to the simple reader.
Let's try creating a custom class for this such that we understand what are the different components here.